libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -0,0 +1,175 @@
+@file:JvmName("NavigationRouteEx")
+
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import java.util.UUID
+
+/**
+ * Wraps a route object used across the Navigation SDK features.
+ *
+ * @param directionsResponse the original response that that returned this route object.
+ * @param routeIndex the index of the route that this wrapper tracks
+ * from the collection of routes returned in the original response.
+ * @param routeOptions options used to generate the [directionsResponse]
+ * @param id unique ID of this route entity used for matching it against additional metadata.
+ * Unique ID doesn't mean that the underlying [directionsRoute] is unique.
+ * Multiple [NavigationRoute]s can represent similar or equal [DirectionsRoute].
+ */
+class NavigationRoute(
+    val directionsResponse: DirectionsResponse,
+    val routeIndex: Int,
+    val routeOptions: RouteOptions,
+    val id: String = UUID.randomUUID().toString()
```

Comment:
What is this `id` used for? Will it be confused with `DirectionsResponse.id`?

Reply:
Tried answering this in https://github.com/mapbox/mapbox-navigation-android/pull/5411#discussion_r809237026.

Reply:
Right, I think it has to be a string so that it's easier to pass to Nav Native and retain ID across platform and native codebases.

cc @DenisPryt @Udumft